File: server/models/payee.rb
DB.create_table? :payees do
  uuid :uuid, primary_key: true
  foreign_key :user_uuid, :users, null: false, type: 'uuid'
  String :name, null: false, unique: true
  DateTime :created_at, null: false
  DateTime :updated_at, null: false
end

class Payee < Sequel::Model
  many_to_one :user, key: :user_uuid
  one_to_many :transactions, key: :payee_uuid
  one_to_many :locations, key: :payee_uuid

  def before_create
    self.updated_at = Time.now
    self.created_at ||= self.updated_at
    super
  end

  def before_update
    self.updated_at ||= Time.now
    super
  end
end

File: packages/server/src/database/entities/Payee.js
import {
  Column,
  CreateDateColumn,
  Entity,
  ManyToOne,
  OneToMany,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm'
import {Location} from './Location'
import {Transaction} from './Transaction'
import {User} from './User'

@Entity()
export class Payee {
  @PrimaryGeneratedColumn('uuid') uuid = undefined

  @ManyToOne(() => User, user => user.payees, {nullable: false})
  user = undefined

  @OneToMany(() => Location, location => location.payee, {
    cascade: true,
    eager: true,
  })
  locations = undefined

  @OneToMany(() => Transaction, transaction => transaction.payee)
  transactions = undefined

  @Column('varchar') name = undefined

  @CreateDateColumn() created_at = undefined

  @UpdateDateColumn() updated_at = undefined
}
